### PR TITLE
changed minbatch logic

### PR DESCRIPTION
**Pull request by DvirDukhan**
this PR changes this `minbatch` logic check for not quitting the batching loop when `minbatchsize` is 0.
This will mimic the 1.0 behavior, as `minbatchsize` is checked outside the batching loop.

### CODE REVIEW

File: src/background_workers.c [hunk 1]
```diff
@@ -238,11 +257,10 @@ static RedisAI_RunInfo **_BGThread_BatchOperations(RunQueueInfo *run_queue_info,
         // there's anything else to batch
         current_batchsize += next_batchsize;
 
-        // If minbatchsize hasn't been set, or if the current batch
-        // size exceeds the minimum batch size already, then we're done.
+        // If batch size exceeds the minimum batch size already, then we're done.
         // Otherwise, if minbatchsize was set and the size wasn't reached,
         // loop until there's something new on the queue
-        if (minbatchsize == 0 || current_batchsize >= minbatchsize) {
+        if (minbatchsize != 0 && current_batchsize >= minbatchsize) {
```
Comment (alonre24) [inline src/background_workers.c:263]: Do we really need to check it? why stopping at minbatchsize if we can extend the batch even further (untill it reaches batchsize)?

File: src/background_workers.c [hunk 2]
```diff
@@ -188,11 +189,29 @@ static RedisAI_RunInfo **_BGThread_BatchOperations(RunQueueInfo *run_queue_info,
         return batch_rinfo;
     }
 
+    // Set the batch to be ready by default (optimistic), change it during run.
+    *batchReady = true;
+    bool timeout = false;
+    // If minbatchsize has been set and we are not past it, we check
+    // if the timeout for min batch has expired, in which case we proceed
+    // anyway
+    if (minbatchsize > 0 && minbatchtimeout > 0) {
```
Comment (alonre24) [inline src/background_workers.c:198]: Maybe we can move the timeout checking to the beginning of the loop (line 214) and avoid the duplication in lines 290-278

**Review by alonre24 — COMMENTED** [review]
Nice! Added 2 small comments for your review

File: src/background_workers.c (hunk 2 above)
Comment (DvirDukhan) [reply src/background_workers.c:198]: if the queue is empty, you need to see that the pending task is not timedout

File: src/background_workers.c (hunk 1 above)
Comment (DvirDukhan) [reply src/background_workers.c:263]: good point

**Review by alonre24 — APPROVED** [review]

**Comment by AvitalFineRedis** [comment]
After the changes in the current PR, we saw that the previous regression (21%) was gone:
image
image

* as we focused on CPU/auto-batching, we should discuss if further tests should be made on GPU and tensor-batching.

**Pull request merged**